homeassistant/components/lgthinq/config_flow.py

```diff
@@ -0,0 +1,146 @@
+"""Config flow for LG ThinQ."""
+
+from __future__ import annotations
+
+import logging
+from typing import Any
+import uuid
+
+import pycountry
+from thinqconnect.thinq_api import ThinQApi
+import voluptuous as vol
+
+from homeassistant.config_entries import ConfigFlow, ConfigFlowResult
+from homeassistant.const import CONF_ACCESS_TOKEN, CONF_COUNTRY, CONF_NAME
+from homeassistant.helpers import config_validation as cv
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+from homeassistant.helpers.selector import (
+    SelectOptionDict,
+    SelectSelector,
+    SelectSelectorConfig,
+    SelectSelectorMode,
+)
+
+from .const import (
+    CLIENT_PREFIX,
+    CONF_CONNECT_CLIENT_ID,
+    DEFAULT_COUNTRY,
+    DOMAIN,
+    THINQ_DEFAULT_NAME,
+    THINQ_PAT_URL,
+    TRANSLATION_ERROR_CODE,
+)
+
+SUPPORTED_COUNTRIES = [
+    SelectOptionDict(value=x.alpha_2, label=x.name) for x in pycountry.countries
+]
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class ThinQFlowHandler(ConfigFlow, domain=DOMAIN):
+    """Handle a config flow."""
+
+    VERSION = 1
+
+    def __init__(self) -> None:
+        """Initialize flow."""
+        super().__init__()
+        self._access_token: str = ""
+        self._country_code: str = DEFAULT_COUNTRY
+        self._entry_name: str = ""
+
+    def _get_default_country_code(self) -> str:
+        """Get the default country code based on config."""
+        country = self.hass.config.country
+        if country is not None:
+            for x in SUPPORTED_COUNTRIES:
+                if x.get("value") == country:
+                    return country
+
+        return DEFAULT_COUNTRY
+
+    async def _validate_and_create_entry(self) -> ConfigFlowResult:
+        """Create an entry for the flow."""
+        connect_client_id: str = f"{CLIENT_PREFIX}-{uuid.uuid4()!s}"
+
+        # To verify PAT, create an api to retrieve the device list.
+        api = ThinQApi(
+            session=async_get_clientsession(self.hass),
+            access_token=self._access_token,
+            country_code=self._country_code,
+            client_id=connect_client_id,
+        )
+        result = await api.async_get_device_list()
+        _LOGGER.debug("validate_and_create_entry: %s", result)
+
+        if result.status >= 400:
```

Comment:
Shouldn't the library raise an error/exception in these cases? Seems a bit odd. As I cannot find the sources of the library, I'm not sure why this is the way it is here.

Reply:
This is just to handle REST API error cases.
Other integrations are doing it in the same way.

Reply:
Removed